chain/epoch-manager/src/adapter.rs

```diff
@@ -721,21 +716,18 @@ impl EpochManagerAdapter for EpochManagerHandle {
         shard_id: ShardId,
     ) -> Result<(ShardId, ShardIndex), Error> {
         let shard_layout = self.get_shard_layout_from_prev_block(prev_hash)?;
-        if self.is_next_block_epoch_start(prev_hash)? {
-            let prev_shard_layout = self.get_shard_layout(&self.get_epoch_id(prev_hash)?)?;
-            if prev_shard_layout != shard_layout {
-                let parent_shard_id = shard_layout.get_parent_shard_id(shard_id)?;
-                assert!(prev_shard_layout.shard_ids().any(|i| i == parent_shard_id),
-                                    "invalid shard layout.  parent_shard_id: {}\nshard_layout: {:?}\nprev_shard_layout: {:?}",
-                                    parent_shard_id,
-                                    shard_layout,
-                                    parent_shard_id
-                            );
-                let parent_shard_index = prev_shard_layout.get_shard_index(parent_shard_id);
-                return Ok((parent_shard_id, parent_shard_index));
-            }
+        let prev_shard_layout = self.get_shard_layout(&self.get_epoch_id(prev_hash)?)?;
+        let is_resharding_boundary =
```

Comment:
nit: we could think of introducing a private function returning `shard_layout`, `prev_shard_layout` and `is_resharding_boundary`

Reply:
Yeah I wanted to do that but it was very annoying to do here because it's a trait impl and I don't really want to add it to the trait itself.